core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -65,6 +68,28 @@ class LogSegmentTest {
     Utils.delete(logDir)
   }
 
+  /**
+   * If the maximum offset beyond index, appended to the log section, it throws LogSegmentOffsetOverflowException
+   */
+  @ParameterizedTest
+  @CsvSource(Array(
```

Comment:
Note that baseOffset can be `Long` but `largestOffset - baseOffset` should be <= `Integer.MaxValue`. This case is missing from our test here.

Could we add the following test cases as well:

baseOffset is a number > Integer.MaxValue, largestOffset is a number > Integer.MaxValue, such that `largestOffset - baseOffset` > Integer.MaxValue. This will throw an error.

Reply:
Thanks, I've added this use case data